pkg/dev/podmandev/reconcile.go

```diff
@@ -118,6 +123,20 @@ func (o *DevClient) reconcile(
 	return nil
 }
 
+// warnAboutK8sResources prints a warning if the Devfile contains a K8s resource that it needs to create on Podman.
+func (o *DevClient) warnAboutK8sResources(devfileObj parser.DevfileObj) {
+	k8sComponents, _ := devfile.GetKubernetesComponentsToPush(devfileObj, true)
```

Comment:
As I understand it, this would include all Devfile components referenced by an `apply` command, regardless of whether `odo dev` uses them or not, right? So if my Devfile contains a `deploy` command with some Kubernetes resources, this will show warnings for such components, even if they will never be used by `odo dev`, no?

Reply:
```
$ odo init --name java-springboot-starter --devfile java-springboot --devfile-registry DefaultDevfileRegistry --devfile-version 2.0.0 --starter springbootproject                                                   

$ ODO_EXPERIMENTAL_MODE=t odo dev --platform=podman
============================================================================
⚠ Experimental mode enabled. Use at your own risk.
More details on https://odo.dev/docs/user-guides/advanced/experimental-mode
============================================================================

  __
 /  \__     Developing using the "java-springboot-starter" Devfile
 \__/  \    Platform: podman
 /  \__/    odo version: v3.5.0
 \__/

↪ Deploying to podman in developer mode
 ✓  Deploying pod [4s]
 ⚠  Kubernetes components are not supported on Podman. Skipping: deploy.
 ✓  Building your application in container (command: build) [46s]
 •  Executing the application (command: run)  ...
 -  Forwarding from 127.0.0.1:40001 -> 8080
 -  Forwarding from 127.0.0.1:40002 -> 5858                                                         
```

I can confirm the behavior, and find that the warning here can be confusing, as, IMO, it makes users think that `odo dev` was about to run the "deploy" component.

Reply:
We also don't support building images for podman as of now, I add adding a warning for those components as well.